etl_entities/hwm/key_value/key_value_hwm.py

```diff
@@ -0,0 +1,144 @@
+#  Copyright 2023 MTS (Mobile Telesystems)
+#
+#  Licensed under the Apache License, Version 2.0 (the "License");
+#  you may not use this file except in compliance with the License.
+#  You may obtain a copy of the License at
+#
+#      http://www.apache.org/licenses/LICENSE-2.0
+#
+#  Unless required by applicable law or agreed to in writing, software
+#  distributed under the License is distributed on an "AS IS" BASIS,
+#  WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+#  See the License for the specific language governing permissions and
+#  limitations under the License.
+from typing import Generic, Optional, TypeVar
+
+from frozendict import frozendict
+from pydantic import Field, validator
+
+from etl_entities.entity import GenericModel
+from etl_entities.hwm.hwm import HWM
+
+KeyValueHWMValueType = TypeVar("KeyValueHWMValueType")
+KeyValueHWMType = TypeVar("KeyValueHWMType", bound="KeyValueHWM")
+
+
+class KeyValueHWM(HWM[frozendict], Generic[KeyValueHWMValueType], GenericModel):
+    """Base key value HWM type
+
+    Parameters
+    ----------
+    name : ``str``
+
+        HWM unique name
+
+    value : ``frozendict[Any, KeyValueHWMValueType]]]`` , default: ``frozendict``
+
+        HWM value
+
+    description : ``str``, default: ``""``
+
+        Description of HWM
+
+    source : Any, default: ``None``
+
+        HWM source, e.g. ``topic`` name
+
+    expression : Any, default: ``None``
+
+        Expression used to generate HWM value, e.g. ``offset``
+
+    modified_time : :obj:`datetime.datetime`, default: current datetime
+
+        HWM value modification time
+    """
+
+    entity: Optional[str] = Field(default=None, alias="topic")
+    # value: frozendict with Any type for keys and KeyValueHWMValueType type for values.
+    # Direct type specification for frozendict contents (e.g., frozendict[KeyType, ValueType])
+    # is supported only from Python 3.9 onwards.
+    value: frozendict = Field(default_factory=frozendict)
+
+    def update(self, new_data: dict) -> "KeyValueHWM[KeyValueHWMValueType]":
```

Comment:
Добавь в начало файла `from __future__ import annotations`, и сможешь ссылаться на класс без оборачивания его в кавычки.

А лучше вообще:
```suggestion
    def update(self: KeyValueHWMType, new_data: dict) -> KeyValueHWMType:
```